src/dstack/_internal/server/app.py

```diff
@@ -166,6 +174,10 @@ def register_routes(app: FastAPI, ui: bool = True):
     app.include_router(gateways.router)
     app.include_router(volumes.root_router)
     app.include_router(volumes.project_router)
+    if FeatureFlags.GATEWAY_IN_SERVER:
+        app.include_router(
+            service_proxy.router, prefix="/gateway/services", tags=["gateway-in-server"]
```

Comment:
/gateway/ in the path would be misleading when the server is proxying the requests. gateway-in-server should remain an internal concept.

Reply:
Also, I remember we agreed that we need two separate concepts: gateways and service proxies, so I'd expect not to see "gateway-in-server" in the code as well. Currently, it's unclear what gateway is since the term is used for different things.

Reply:
Gateway is a component of dstack used to access services and models. It can run in two modes: in-server and remotely. Each mode has it's own specifics, e.g. remote gateways need to be created, while gateway-in-server exists always, remote gateways can serve services at subdomains, while gateway-in-server can't, etc.

I think this is easier for users to understand than a new term. "Service proxy" isn't a suitable term because this component will also be used for LLMs, not just services. So it will need a more generic name, like "Proxy". And having two similar terms like "Gateway" and "Proxy" can cause confusion.